twilio-authenticator-sample/src/main/java/com/twilio/authenticatorsample/appdetail/AppsAdapter.java

```diff
@@ -57,31 +57,45 @@ public int getItemCount() {
         return apps == null ? 0 : apps.size();
     }
 
-    public void addApp(App app) {
-        apps.add(app);
+    public void addApps(List<App> apps) {
+        this.apps.addAll(apps);
         notifyDataSetChanged();
     }
 
-    public void removeApp(String appId) {
+    private void removeApp(Long appId) {
         Iterator<App> iterator = apps.iterator();
         while (iterator.hasNext()) {
             App app = iterator.next();
-            if (app.getId().equals(appId)) {
+            if (app.getId() == appId) {
                 iterator.remove();
                 break;
             }
         }
+    }
+
+    public void removeApps(List<Long> appIds) {
+        for (Long appId: appIds) {
+            removeApp(appId);
+        }
+
         notifyDataSetChanged();
+
     }
 
-    public void updateApp(App authenticatorToken) {
-        removeApp(authenticatorToken.getId());
-        addApp(authenticatorToken);
+    public void updateApps(List<App> apps) {
+
+        for (App app: apps) {
+            removeApp(app.getId());
+            apps.add(app);
```

Comment:
can we use `this.` here for clarity? It's confusing that the parameter and property are equally named.

Reply:
fixed